### PR TITLE
fix: Add vector database doc

### PR DESCRIPTION
**Pull request by HaoXuAI**
# What this PR does / why we need it:

Add the instruction document for vector database
# Which issue(s) this PR fixes:

# Fixes

### CODE REVIEW

File: docs/reference/alpha-vector-database.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+offline_store:
+  type: file
+entity_key_serialization_version: 2
+```
+Run the following command to apply the feature store configuration:
+
+```shell
+feast apply
+```
+
```
Comment (franciscojavierarceo) [inline docs/reference/alpha-vector-database.md:59]:
```suggestion

Note that when you run `feast apply` you are going to apply the following Feature View that you will use for retrieval later:
```python
city_embeddings_feature_view = FeatureView(
    name="city_embeddings",
    entities=[item],
    schema=[
        Field(name="Embeddings", dtype=Array(Float32)),
    ],
    source=source,
    ttl=timedelta(hours=2),
)
```

```

File: docs/reference/alpha-vector-database.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+Then run the following command to materialize the data to the online store:
+
+```shell
+!feast materialize 2024-04-01T00:00:00 2024-04-17T00:00:00
```
Comment (franciscojavierarceo) [inline docs/reference/alpha-vector-database.md:63]: Would suggest the below so time doesn't impact this
```suggestion
!CURRENT_TIME=$(date -u +"%Y-%m-%dT%H:%M:%S")
!feast materialize-incremental $CURRENT_TIME
```

**Review by franciscojavierarceo — CHANGES_REQUESTED** [review]
Nice! Overall lgtm. I'll try to clean up the full demo this week.

CC @jeremyary can we include this in 37.1?

File: docs/reference/alpha-vector-database.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  type: file
+entity_key_serialization_version: 2
+```
+Run the following command to apply the feature store configuration:
+
+```shell
+feast apply
+```
+
+Then run the following command to materialize the data to the online store:
```
Comment (franciscojavierarceo) [inline docs/reference/alpha-vector-database.md:60]:
```suggestion
Then run the following command in the terminal to materialize the data to the online store:
```

**Comment by jeremyary** [comment]
> Nice! Overall lgtm. I'll try to clean up the full demo this week.
> 
> CC @jeremyary can we include this in 37.1?

hmm, with it being just a doc, I think it may sync to gitbook with just a cherry-pick to v0.37-branch and maybe a nudge on gitbook side. That said, we've got a lot of goodies going in this week so I was considering a 0.38 cut before too awful long

File: docs/reference/alpha-vector-database.md (hunk 1 above)
Comment (HaoXuAI) [reply docs/reference/alpha-vector-database.md:59]: make sense. Updated the doc

**Comment by HaoXuAI** [comment]
```
Run aws-actions/amazon-ecr-login@v1
Warning: Your docker password is not masked. See https://github.com/aws-actions/amazon-ecr-login#docker-credentials for more information.
Error: User: arn:aws:iam::***:user/feast is not authorized to perform: ecr:GetAuthorizationToken on resource: * because no identity-based policy allows the ecr:GetAuthorizationToken action
```
@jeremyary looks like there is a aws auth issue.

File: docs/reference/alpha-vector-database.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+Run the following command in terminal to apply the feature store configuration:
+
+```shell
+feast apply
+```
+
+Note that when you run `feast apply` you are going to apply the following Feature View that we will use for retrieval later:  
+
+```python
+city_embeddings_feature_view = FeatureView(
```
Comment (franciscojavierarceo) [inline docs/reference/alpha-vector-database.md:63]: This section didn't format properly.

Comment (HaoXuAI) [reply docs/reference/alpha-vector-database.md:63]: good catch. updated

**Review by franciscojavierarceo — APPROVED** [review]

**Pull request merged**